From: Rob Evers <revers@redhat.com>
Date: Fri, 10 Jul 2009 14:17:32 -0400
Subject: [scsi] lpfc: fix ctx_idx increase and update version
Message-id: 20090710181301.4580.4650.sendpatchset@localhost.localdomain
O-Subject: [RHEL5.4 PATCH 2/3 v3] lpfc: patch to fix ctx_idx inc. and rollover in [1/3], version 8.2.0.48.1p.
Bugzilla: 509010

https://bugzilla.redhat.com/show_bug.cgi?id=509010

Update lpfc from 8.2.0.48 to 8.2.0.48.1p

--

diff --git a/drivers/scsi/lpfc/lpfc_ioctl.c b/drivers/scsi/lpfc/lpfc_ioctl.c
index 9f45bff..132dbac 100644
--- a/drivers/scsi/lpfc/lpfc_ioctl.c
+++ b/drivers/scsi/lpfc/lpfc_ioctl.c
@@ -2015,8 +2015,8 @@ lpfcdfc_ct_unsol_event(struct lpfc_hba * phba,
 
 		mutex_lock(&lpfcdfc_lock);
 		if (phba->sli_rev == LPFC_SLI_REV4) {
-			evt_dat->immed_dat = dfchba->ctx_idx % 64;
-			dfchba->ctx_idx += dfchba->ctx_idx++ % 64;
+			evt_dat->immed_dat = dfchba->ctx_idx;
+			dfchba->ctx_idx = (dfchba->ctx_idx + 1) % 64;
 			dfchba->ct_ctx[evt_dat->immed_dat].oxid =
 						piocbq->iocb.ulpContext;
 			dfchba->ct_ctx[evt_dat->immed_dat].SID =
diff --git a/drivers/scsi/lpfc/lpfc_version.h b/drivers/scsi/lpfc/lpfc_version.h
index e0ead14..cb0e96e 100644
--- a/drivers/scsi/lpfc/lpfc_version.h
+++ b/drivers/scsi/lpfc/lpfc_version.h
@@ -18,7 +18,7 @@
  * included with this package.                                     *
  *******************************************************************/
 
-#define LPFC_DRIVER_VERSION "8.2.0.48"
+#define LPFC_DRIVER_VERSION "8.2.0.48.1p"
 
 #define LPFC_DRIVER_NAME		"lpfc"
 #define LPFC_SP_DRIVER_HANDLER_NAME	"lpfc:sp"
